src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,179 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * The TaskList class is a list of {@link Task}s.
+ * It supports the basic operations of manipulating tasks and answering queries.
+ */
+public class TaskList {
+    private List<Task> tasks;
+
+    /**
+     * Instantiates an empty list of {@link Task}s.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Instantiates a new TaskList with a provided list of {@link Task}s.
+     * @param tasks A list of {@link Task}s.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a {@link Task} to the end of the list.
+     * @param task A task to be added to the list.
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Marks a {@link Task} at a certain index (starting from 1) as completed.
+     * @param index The index of the task to be marked as completed.
+     */
+    public void markDone(int index) {
+        tasks.get(index - 1).markAsDone();
+    }
+
+    /**
+     * Returns a {@link Task} in the list based on the given index.
+     * @param index Index of the task in the list.
+     * @return The task at the provided index.
+     */
+    public Task getTask(int index) {
+        return tasks.get(index - 1);
+    }
+
+    /**
+     * Returns the current number of {@link Task}s in the list.
+     * @return The current size of the list.
+     */
+    public int getSize() {
+        return tasks.size();
+    }
+
+    /**
+     * Returns the list of all the {@link Task}s in the current list.
+     * @return The list of tasks.
+     */
+    public List<Task> getTasks() {
+        return new ArrayList<>(tasks);
+    }
+
+    /**
+     * Returns all the {@link Task}s formatted properly.
+     * @return A formatted String of all the {@link Task}s.
+     */
+    public String listTasks() {
+        StringBuilder output = new StringBuilder();
+        for (int i = 0; i < tasks.size(); i++) {
+            output.append(i + 1)
+                    .append(".")
+                    .append(tasks.get(i))
+                    .append(i == tasks.size() - 1 ? "" : "\n");
+        }
+        return output.toString();
+    }
+
+    /**
+     * Returns all the {@link Task}s that happen on or due at a specific date.
+     * @param date The date to search for.
+     * @return A formatted String of all the tasks that happen on or due at the provided date.
+     */
+    public String showTasksOnDate(LocalDate date) {
```

Comment:
thought this method was really cool and unique!!

Reply:
Thanks haha!